practice-two/src/components/FormInput/FormInput.jsx

```diff
@@ -0,0 +1,60 @@
+import { useState } from 'react';
+import './FormInput.css';
+import PropTypes from 'prop-types';
+import { MESSAGES, REGEX } from '@constants';
+
+const FormInput = (props) => {
+  const [focused, setFocused] = useState(false);
+  const { label, errorMessage, onChange, ...inputProps } = props;
+
+  const handleFocus = () => {
+    setFocused(true);
+  };
+
+  return (
+    <div className='form__input'>
+      <label className='input__label'>{label}</label>
+      <input
+        className='input__field'
+        {...inputProps}
+        onChange={onChange}
+        onBlur={handleFocus}
+        // eslint-disable-next-line react/no-unknown-property
+        focused={focused.toString()}
+      />
+      <span className='error__message'>{errorMessage}</span>
+    </div>
+  );
+};
+
+FormInput.propTypes = {
+  label: PropTypes.string,
+  placeholder: PropTypes.string,
+  type: PropTypes.oneOf(['text', 'tel', 'email']),
+  errorMessage: PropTypes.oneOf([
```

Comment:
how about errorMessage empty string

Reply:
- I added case empty string error to messages constants
![image](https://github.com/datdragon777/React-Training/assets/74806826/af1de55c-e8c6-4001-a819-ae3eae636fe9)

- When I set error empty string:
![image](https://github.com/datdragon777/React-Training/assets/74806826/5e2d22f4-4b82-4971-a121-fd07dfbd1fdc)

- When I have error message:
![image](https://github.com/datdragon777/React-Training/assets/74806826/328066a6-60e7-4134-a4ac-04d91f0bf13e)